adam/perception/perception_graph.py

```diff
@@ -2784,22 +2803,19 @@ def translate_frames(
         # First, we translate each of the two frames into PerceptionGraphs independently.
         # The edges of each graph are marked with the appropriate "temporal specifier"
         # which tells whether they belong to the "before" frame or the "after" frame.
-        before_frame_graph = (
-            self.translate_frame(perceptual_representation.frames[0])
-            .copy_with_temporal_scopes([TemporalScope.BEFORE])
-            .copy_as_digraph()
+        before_frame_graph = PerceptionGraph.copy_digraph_with_temporal_scopes(
```

Comment:
Here (and for `AFTER` below), do we need to copy the result of `_translate_frame`, since there is no other reference to the returned `DiGraph`?

Reply:
That is, could we make an in-place `add_temporal_scopes_to_edges` function instead? The copy still take ~1/4 of this function's time in the profile.

Reply:
Yes, I think so.

Reply:
I changed it to add the temporal scopes in-place.